msal4j-sdk/src/integrationtest/java/com.microsoft.aad.msal4j/AcquireTokenSilentIT.java

```diff
@@ -69,6 +69,10 @@ void acquireTokenSilent_LabAuthority_TokenNotRefreshed(String environment) throw
         // Check that access and id tokens are coming from cache
         assertEquals(result.accessToken(), acquireSilentResult.accessToken());
         assertEquals(result.idToken(), acquireSilentResult.idToken());
+        assertEquals(TokenSource.IDENTITY_PROVIDER, result.metadata().tokenSource());
+        assertEquals(CacheRefreshReason.NOT_APPLICABLE, result.metadata().cacheRefreshReason());
+        assertEquals(TokenSource.CACHE, acquireSilentResult.metadata().tokenSource());
+        assertEquals(CacheRefreshReason.NOT_APPLICABLE, acquireSilentResult.metadata().cacheRefreshReason());
```

Comment:
Please add more tests. Consider adding these assertions to all integrationt tests, not just public client ones. Client_credentials is of highest priority. 

In particular:

- client_credentials + all combos of CacheRefreshReason (especially PROACTIVE_REFRESH)
- client_credentials + claims